File: 백준/Silver/18258. 큐 2/큐 2.java
import java.io.*;
import java.util.*;

public class Main {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringBuilder sb = new StringBuilder();

		int t = Integer.parseInt(br.readLine());

			Deque<Integer> q = new LinkedList<>();
		while (t-- > 0) {
			String s = br.readLine();

			StringTokenizer st = new StringTokenizer(s);
			String input = st.nextToken();

			if (input.equals("push")) {
				int num = Integer.parseInt(st.nextToken());
				q.add(num);
				continue;
			} else if (input.equals("pop")) {
				if (q.isEmpty())
					sb.append("-1");
				else {
					sb.append(q.peek());
					q.remove();
				}
			} else if (input.equals("size")) {
				sb.append(q.size());
			} else if (input.equals("empty")) {
				if (q.isEmpty())
					sb.append("1");
				else
					sb.append("0");
			} else if (input.equals("front")) {
				if (q.isEmpty())
					sb.append("-1");
				else
					sb.append(q.peek());
			} else {
				if (q.isEmpty())
					sb.append("-1");
				else
					sb.append(q.peekLast());
			}
			sb.append("\n");
		}

		System.out.println(sb);
	}
}

File: 백준/Silver/18258. 큐 2/큐 2.cc
#include <iostream>
#include <queue>
#include <string>

using namespace std;

int n;
queue<int> q;
string cmd;
int num;

int main() {
    ios_base::sync_with_stdio(0); cin.tie(0);

    cin >> n;

    while (n--) {
        cin >> cmd;
        if (cmd == "push") {
            cin >> num;
            q.push(num);
            continue;
        } else if (cmd == "pop") {
            if (!q.empty()) {
                cout << q.front();
                q.pop();
            }
            else cout << -1;
        } else if (cmd == "size") {
            cout << q.size();
        } else if (cmd == "empty") {
            if (q.empty()) cout << 1;
            else cout << 0;
        } else if (cmd == "front") {
            if (!q.empty()) cout << q.front();
            else cout << -1;
        } else if (cmd == "back") {
            if (!q.empty()) cout << q.back();
            else cout << -1;
        }
        cout << "\n";
    }

    return 0;
}